mesonbuild/wrap/wrap.py

```diff
@@ -333,6 +346,13 @@ def apply_patch(self):
                 shutil.unpack_archive(path, workdir)
                 self.copy_tree(workdir, self.subdir_root)
 
+    def apply_patch_file(self, filename, strip):
+        mlog.log('Applying patch ' + filename)
+        full_path = os.path.join(self.subdir_root, filename)
+        p, o, e = Popen_safe(['patch', '-p' + strip, '-f', '-i', full_path], cwd=self.dirname)
```

Comment:
This is not acceptable, use of shell commands is not permitted, because they don't exist on Windows. Only pure Python is supported and since this functionality is not in the stdlib, it would mean implementing `patch` from scratch. At that point it becomes easier to just do the current unzip thingy.

Reply:
That's user choice, we invoke tones of command lines that are not avaible on Windows, the whole GNOME module for instance.

Reply:
also git

Reply:
TIL, git for windows installs patch.exe but not in PATH by default.

Reply:
oh, even better, `git apply foo.patch` even works when it's not in a git repository. It's a plain alias to patch command it seems. We could use that.